fastestimator/trace/adapt/terminate_on_nan.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Set, Union
+
+import numpy as np
+import tensorflow as tf
+import torch
+
+from fastestimator.schedule.schedule import get_current_items
+from fastestimator.trace.trace import Trace
+from fastestimator.util.data import Data
+from fastestimator.util.traceability_util import traceable
+from fastestimator.util.util import to_set
+
+
+@traceable()
+class TerminateOnNaN(Trace):
+    """End Training if a NaN value is detected. By default (inputs=None) it will monitor all loss values at the end
```

Comment:
First Line of doc string is brief summary (can't be more than 1 line). Extra details go in subsequent paragraph

Reply:
I will update the doc string. Thanks.